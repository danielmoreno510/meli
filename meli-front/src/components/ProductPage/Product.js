import PropTypes from "prop-types";

import { formatCurrency } from "../../utils/currencyUtils";
import "./styles.scss";

const Product = ({ product }) => (
  <div>
    <div className="product-container">
      <img
        className="product-image"
        src={product.picture}
        alt={product.title}
      />
      <div className="product-section">
        <div className="product-condition">
          {product.condition} - {product.sold_quantity} vendidos
        </div>
        <h1 className="product-title">{product.title}</h1>
        <div className="product-amount">
          {formatCurrency(product.price.currency, product.price.amount)}
        </div>
        <div className="product-button">Comprar</div>
      </div>
    </div>
    <h2 className="product-description-title">Descripcion del producto</h2>
    <div className="product-description">{product.description}</div>
  </div>
);

Product.propTypes = {
    product: PropTypes.object,
  };

  export default Product