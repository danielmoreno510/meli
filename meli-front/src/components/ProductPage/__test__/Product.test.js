import React from "react";
import { shallow } from "enzyme";

import Product from "../Product";

describe("ProductPage component", () => {
  const testProps = {
    product: {
      id: "MLA916044386",
      thumbnail: "image.png",
      title: "iPhone 11 64 Gb Blanco",
      price: { currency: "ARS", amount: "123456" },
    },
  };
  describe("WHEN the Product component is render", () => {
    const wrapper = shallow(<Product {...testProps} />);

    it("THEN should display Product component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
