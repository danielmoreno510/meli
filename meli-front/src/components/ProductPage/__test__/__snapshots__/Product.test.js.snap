// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage component WHEN the Product component is render THEN should display Product component 1`] = `
<div
  className="product-container"
>
  <div
    className="product-content"
  >
    <img
      alt="iPhone 11 64 Gb Blanco"
      className="product-image product-responsive"
    />
    <div
      className="product-section product-responsive"
    >
      <div
        className="product-condition"
      >
         - 
         vendidos
      </div>
      <h1
        className="product-title"
      >
        iPhone 11 64 Gb Blanco
      </h1>
      <div
        className="product-amount"
      >
        $ 123,456
      </div>
      <div
        className="product-button"
      >
        <div
          className="product-button-content"
        >
          Comprar
        </div>
      </div>
    </div>
  </div>
  <h2
    className="product-description-title"
  >
    Descripcion del producto
  </h2>
  <div
    className="product-description product-responsive"
  />
</div>
`;
